doc/build-netbsd.md

```diff
@@ -0,0 +1,49 @@
+NetBSD build guide
+======================
+(updated for NetBSD 7.0)
+
+This guide describes how to build bitcoind and command-line utilities on NetBSD.
+
+This guide does not contain instructions for building the GUI.
+
+Preparation
+-------------
+
+You will need the following modules, which can be installed via pkgsrc or pkgin:
+
+```
+autoconf
+automake
+boost
+db4
+git
+gmake
+libevent
+libtool
+python27
+```
+
+Download the source code:
+```
+git clone https://github.com/bitcoin/bitcoin
+```
+
+See [dependencies.md](dependencies.md) for a complete overview.
+
+### Building Bitcoin Core
+
+**Important**: Use `gmake` (the non-GNU `make` will exit with an error).
+
+With wallet:
+```
+./autogen.sh
+./configure CPPFLAGS="-I/usr/pkg/include -DOS_NETBSD" LDFLAGS="-L/usr/pkg/lib" BOOST_CPPFLAGS="-I/usr/pkg/include" BOOST_LDFLAGS="-L/usr/pkg/lib"
```

Comment:
Could you please add this: https://github.com/bitcoin/bitcoin/pull/12125#issuecomment-356472816

Then -DOS_NETBSD shouldn't be needed, which I suspect is enough to fix the upstream package.

Reply:
@theuni As I recall, the -DOS_NETBSD flag is needed because the Makefiles and various other scripts seem to have problems detecting the OS reliably, and sometimes the issue is with case-sensitivity.  The inclusion of this flag results in a successful build/compile, but if it's possible to update the needed files to eliminate this then I think that would be excellent.

Reply:
Yes, it's needed for leveldb. We detect it for other OSs, there's no need to tream netbsd differently in this regard.